src/main/java/com/google/netpcapanalysis/dao/FrequencyDaoImpl.java

```diff
@@ -84,11 +80,44 @@ public String getMyIP() {
 
   /* Fills out finalMap with frequencies based on IP address only (not protocol). */
   private void processData(){
-    finalMap = new HashMap<String, PCAPdata>();
+    LinkedHashMap<String, Integer> hm = getUniqueOutIPs();
```

Comment:
Do you really need a linkedHashMap that offers ordering?

Reply:
Changed the structure of the code so that the backend now doesn't limit the number of addresses passed into JavaScript visualization; the JavaScript visualization only shows the first `n` addresses in the linkedHashMap, which are the `n` most frequent connections